playground/CustomResources/CustomResources.AppHost/TestResource.cs

```diff
@@ -0,0 +1,73 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Hosting.Dashboard;
+using Aspire.Hosting.Lifecycle;
+
+static class TestResourceExtensions
+{
+    public static IResourceBuilder<TestResource> AddTestResource(this IDistributedApplicationBuilder builder, string name)
+    {
+        builder.Services.AddLifecycleHook<TestResourceLifecycleHook>();
+
+        var rb = builder.AddResource(new TestResource(name))
+                      .WithDashboardState(state => state with
```

Comment:
I don't like dashboard in the name. It isn't dashboard state, it's state that happens to be shown in the dashboard.

Reply:
I guess custom resources like this. are local development only.

Reply:
Right, @mitchdenny had the same feedback though. I'm open to suggestions!